src/main/php/PHPMD/Renderer/DatatablesRenderer.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/**
+ * Initial version created on: 03.12.2020 14:31
+ *
+ * This renderer uses parts of the HTMLRenderer by Premysl Karbula
+ *
+ * @author  Tebin Ulrich <info@tebinulrich.de>
+ * @license http://www.opensource.org/licenses/bsd-license.php BSD License
+ */
+
+namespace PHPMD\Renderer;
+
+use PHPMD\Renderer\HTMLRenderer;
+use PHPMD\Report;
+use SplFileObject;
+
+/**
+ * Class DatatablesRenderer
+ *
+ * @package PHPMD\Renderer
+ */
+class DatatablesRenderer extends HTMLRenderer
+{
+    /**
+     * Writes the beginning of the report, including css and js
+     */
+    public function start() {
+        $writer = $this->getWriter();
+        
+        $cssDatatables = file_get_contents(__DIR__ . '/DatatablesRenderer/jquery.dataTables.min.css');
+        $cssLocal      = file_get_contents(__DIR__ . '/DatatablesRenderer/DatatablesRenderer.css');
+        $jsjQuery      = file_get_contents(__DIR__ . '/DatatablesRenderer/jquery-3.5.1.js');
+        $jsDatatables  = file_get_contents(__DIR__ . '/DatatablesRenderer/jquery.dataTables.min.js');
+        
+        $writer->write(
+            "<!DOCTYPE html>
+<head>
+    <title>PHPMD Datatables Report</title>
+    <link rel='shortcut icon' href='data:image/x-icon;,' type='image/x-icon'>
+    <style>
+    {$cssDatatables}
+    {$cssLocal}
+    </style>
+    <script  type='text/javascript'>
+    {$jsjQuery}
+    {$jsDatatables}
+    </script>
+</head>
+<body>" . PHP_EOL
+        );
+    }
+    
+    /**
+     * @param $violation
+     *
+     * @return string
+     */
+    public function getPreview($violation) {
+        $previewLines = self::getLineExcerpt($violation->getFileName(), $violation->getBeginLine(), 2);
+        
+        $excerptHtml = '<ul class="code">';
+        foreach ($previewLines as $line => $code) {
+            $class       = $line === $violation->getBeginLine() ? " class='hlt'" : null;
+            $codeHtml    = htmlspecialchars($code);
+            $excerptHtml .= "<li{$class}><div class='no'>{$line}</div><div class='cd'>{$codeHtml}</div></li>";
+        }
+        $excerptHtml .= '</ul>';
+        
+        return $excerptHtml;
+    }
+    
+    /**
+     * Writes the report
+     *
+     * @param Report $report
+     */
+    public function renderReport(Report $report) {
+        $violations = $report->getRuleViolations();
+        $writer     = $this->getWriter();
+        
+        $writer->write(
+            "
+            <h2>PHPMD Datatables Report</h2>
+            <div class='info'>generated in <b>" . ($report->getElapsedTimeInMillis(
+                ) / 1000) . "</b> seconds.<br />A total of <b>" . count($violations) . "</b> violations have been detected.</div>
```

Comment:
Weird to have a line break in the method brackets
```suggestion
            <div class='info'>generated in <b>" 
            . ($report->getElapsedTimeInMillis() / 1000) 
            . "</b> seconds.<br />A total of <b>" . count($violations) 
            . "</b> violations have been detected.</div>
```

Reply:
@tebeso I meant the 
```
$report->getElapsedTimeInMillis(
)
``` 
part should be `$report->getElapsedTimeInMillis()`